core/metacling/test/TClingTests.cxx

```diff
@@ -239,6 +239,21 @@ TEST_F(TClingTests, GetSharedLibDeps)
 }
 #endif
 
+// Check that a warning message is generated when using auto-injection.
+TEST_F(TClingTests, WarningAutoInjection)
+{
+
+   testing::internal::CaptureStderr();
+   gInterpreter->ProcessLine("/* no auto */ t = new TTree(\"tree\",\"tree\");");
+   std::cerr.flush();
+
+   std::string warningMessage = testing::internal::GetCapturedStderr();
```

Comment:
We might need to capture `Stdout`, too.

Reply:
I tried it on my local windows machine. `GetCapturedStdout()` doesn't catch the message either.

Reply:
I was wondering if it is ok to disable this test on windows, catching the warning message doesn't seem to work on windows. cc: @hahnjo @bellenot

Reply:
Did you "just" try `CheckDiagsRAII`? That calls `gInterpreter->ReportDiagnosticsToErrorHandler()` which should redirect all diagnostics in a platform-independent way...